File: 704-binary-search/704-binary-search.cpp
class Solution {
public:
  int search(vector<int>& nums, int target){
      
      int size = nums.size();
      
  if( size == 0)
    return -1;

  int left = 0;
  int right = size;
      
  while(left < right)
  {
    
    int mid = left + (right - left) / 2;
      
    if(nums[mid] == target)
    return mid; 
    else if( target >nums[mid])
    left = mid + 1; 
    else 
    right = mid; 
  }

  
  // if(left != nums.size() && nums[left] == target) return left;
      
  return -1;
}
};

File: 704-binary-search/704-binary-search.java
class Solution {
    public int search(int[] nums, int target) {
     
        int n = nums.length;
        // int mid = n/2;
        int l=0, r =n-1;
        
        while(l<=r)
        {
            int mid = (l+r)/2;
            if(nums[mid]==target)
                return mid;
            else if(target < nums[mid])
                r=mid-1;
            else
                l=mid+1;
        }
        
        return -1;
        
    }
}